### ISSUE TITLE
24x7 IntelliSense only at the end of words

### ISSUE THREAD

**Issue by user3323** [Microsoft/vscode#32201]
- VSCode Version: 1.14.2
- OS Version: win7x64

Steps to Reproduce:

You have "javascript.json" file with your snippets, and for example there is console.log snippet:

```
"CONSOLE.LOG": {
  "prefix": "c",
  "body": [
    "console.log"
  ],
  "description": "CONSOLE.LOG"
}
```

So, here is letter "c" + Enter = console.log
Very short and useful, but, if in my code there will be for example "**alert**", if I will select it and type letter "c" - there will be no snippets list.

It requires to remove the "alert" before typing "c" with showing snippets.

Should be fixed.

GIF:

![alert - console](https://user-images.githubusercontent.com/24613274/29146077-392772b8-7d68-11e7-82b0-66ddfab55ced.gif)

**Comment by jrieken** [comment]
Yes and that is how it is currently implemented. There is two ways to get IntelliSense, explicit with press Ctrl+Space and implicit while typing. The latter we call 24x7 IntelliSense to ensure it's not annoying we decided that it only shows when typing at the end of a word. 

Most users are are OK with this and we cannot change the default. However we can think about a setting for this.

**Comment by jrieken** [comment]
So, instead of the boolean configuration value we have today, (`"editor.quickSuggestions": true`) we could have an enum, ala `endOfWord`, `insideWords`, `everywhere`.

**Comment by user3323** [comment]
@jrieken, nice. But my question is - would this feature do something with "snippets after dot" issue? (https://github.com/Microsoft/vscode/issues/32231)

**Comment by jrieken** [comment]
> would this feature do something with "snippets after dot" issue? 

Yeah, that would probably `everywhere`

**Comment by user3323** [comment]
@jrieken, thanks for your hard work!

**Comment by user3323** [comment]
@jrieken, hello!

Is there any advance on the **snippets after dot** issue?

**Comment by user3323** [comment]
@jrieken, are you there?

Are there any news?

**Comment by jrieken** [comment]
> Are there any news?

Surprise, surprise but if there would be any news they would appear here.

**Comment by user3323** [comment]
@jrieken  I created this issue like 45 days ago, and still nothing? Maybe you need some more helpful information about this issue?

**Comment by user3323** [comment]
@jrieken, Johannes, are you there? :)

Well, there are some more information on this issue, maybe it will help somehow:

User's custom snippets appear in the list right after typing dot - if the variable wasn't declared at all:
![without variable](https://user-images.githubusercontent.com/24613274/31605968-658b7218-b270-11e7-804f-f1d3cb48c385.gif)

--

Or if the variable was just declared, without any assignment:
![var test](https://user-images.githubusercontent.com/24613274/31606054-bba8095e-b270-11e7-9940-0bdda06cf8cd.gif)

--

If the variable has an assignment - the user's custom snippets will appear in the list only if there are no basic snippets found in the list:
![when the list is empty](https://user-images.githubusercontent.com/24613274/31606516-f8725f14-b271-11e7-9527-b074b6f1fba7.gif)

Cannot believe that in the whole VS Code dev's team - no one can handle and fix this issue...

**Comment by user3323** [comment]
@jrieken 
using Insiders build, and after one of the latest updates, this issue's become even worst!

![ael after dot](https://user-images.githubusercontent.com/24613274/33723809-5148512a-db76-11e7-9e75-e6721f810a6b.gif)

Well done, guys! Thanks for the fixing it ... inversely.

**Comment by jrieken** [comment]
I have locked this issue and deleted some comments that didn't add value to understanding and discussion of the desired feature. While the request is entirely fair and justified there no reason to constantly harness folks. We do prioritise issues by up-votes (👍-reaction) and this issue hasn't seen any yet. I will unlock this issue when emotions calm down again. Thanks for understanding.

**Comment by jrieken** [comment]
This is now actually fixed - most likely via fixing #53945


![sep-12-2018 15-20-55](https://user-images.githubusercontent.com/1794099/45427657-aa6b1580-b69f-11e8-9bfe-3b420962f5a0.gif)
